### DISCUSSION TITLE
V-autocomplete select its text on focus

### DISCUSSION THREAD

**Discussion by stingwear** [vuetifyjs/vuetify#13562] [Q&A]
Hi all.

We have detected strange behavior with the v-automplete component. When it receives the focus, by code, it select all its text:

<img width="469" alt="cap01" src="https://user-images.githubusercontent.com/23191019/116207003-b23c3300-a73f-11eb-81b1-6ec002f95faa.png">

This is very useful in almost every environments except in Safari on iOS, because each time you click on the component and the text is selected, appears the text options (copy, cut, etc) and this is very annoying for users.

It's possible to disable the text selection by parameters? Any help is welcome.

**Comment by baadev** [comment]
is there any resolution or workaround of this?

**Reply by masteroleary** [reply]
HELLO...... ANYBODY MONITORING THESE?

**Comment by Razvan-BT** [comment]
Why i use v-autocomplete? I can use directly v-select, and focus is disabled, and work perfect

**Reply by kieuminhcanh** [reply]
@Razvan-BT 
VAutocomplete is use in case these are large data items. E.g: Search and select an icon from thousands of icons.

**Comment by shiouming** [comment]
It not only affects iphone users, it happens on Android browser too. Perhaps test coverage for mobile web wasn't as solid as desktop web?

**_Good news_**:
I'm not able to reproduce this in v3 docs sample. I supposed the docs is using `v3.3.6`
https://vuetifyjs.com/en/components/autocompletes/

**_Sad news_**:
It's reproducible in latest v2 docs sample (`v2.6.15`?). My project can't afford to upgrade yet.
https://v2.vuetifyjs.com/en/components/autocompletes/

@stingwear looks like the code was introduced simply to make it visually highlighted, not a bad thing by itself:
https://github.com/vuetifyjs/vuetify/commit/17c4d2eea0cea3d1007ab87b810b553988795122

**Comment by shiouming** [comment]
Since the auto input text selection feature is no longer happening in v3, I've created a pull request for v2 to deal with the reported UX issue.
https://github.com/vuetifyjs/vuetify/pull/18090

**Comment by colewerner** [comment]
I actually need the auto selection of text on focus and in the latest version it's not doing it.  I tried doing it programmatically with:

```
this.$refs.myAutoComplete.focus();
this.$refs.myAutoComplete.select();
```

This does not work though.  The focus works but not the select.

**Reply by colewerner** [reply]
Fwiw I asked this question on stackoverflow and a solution is here also to the problem I describe above:

https://stackoverflow.com/questions/78020956/vuetify-3-v-autocomplete-select-all-text-on-focus

**Comment by qxygene1** [comment]
any updates?
